lib/modules/swarm/process.js

```diff
@@ -12,21 +12,43 @@ class SwarmProcess extends ProcessWrapper {
     this.blockchainConfig = options.blockchainConfig;
     this.cors = options.cors;
     this.command = this.storageConfig.swarmPath || 'swarm';
+    this.defaultAccount = options.defaultAccount;
   }
 
   startSwarmDaemon() {
     const self = this;
-    if (!this.storageConfig.account || !this.storageConfig.account.address || !this.storageConfig.account.password) {
+    let bzzaccount;
+    let password;
+    // use our storage config address/password if we have it
+    if (this.storageConfig.account && this.storageConfig.account.address && this.storageConfig.account.password) {
+      bzzaccount = this.storageConfig.account.address;
+      password = fs.dappPath(this.storageConfig.account.password);
+    }
+    // default to our blockchain config account, or our default account
+    else if (!this.blockchainConfig.isDev &&
+      this.blockchainConfig.mineWhenNeeded &&
```

Comment:
I don't think `mineWhenNeeded` is a necessary check for the accounts no?

Reply:
Realistically, it is the only time that the `defaultAccount` is set, but yes, you're right. This has been removed and it simply checked for the blockchain address/password as the backup now.